mlflow/pyfunc/__init__.py

```diff
@@ -847,7 +862,23 @@ def predict(*args):
         else:
             return result[result.columns[0]]
 
-    return pandas_udf(predict, result_type)
+    udf = pandas_udf(predict, result_type)
+    udf.metadata = model_metadata
+
+    @functools.wraps(udf)
+    def udf_with_default_cols(*args):
+        if len(args) == 0:
+            input_schema = model_metadata.get_input_schema()
+
+            if input_schema and input_schema.has_input_names():
+                input_names = input_schema.input_names()
+                return udf(*input_names)
+            else:
+                return udf()
```

Comment:
can we check if the model has input schema with no column names? and raise an error in that case?

Reply:
Actually, would it be fair to say that we do not support models with empty argument list that have no signature? And just raise an error whenever there is an empty argument list but no signature (or one with unnamed columns)? 

I think this would lead to much better user experience for the common case. It is possible there is not a single model with no arguments that is being deployed this way.